docs/deployment/deployment-checklist.md

```diff
@@ -0,0 +1,258 @@
+
+<dl>
+    <dt>Reading metadata of the repositories you select</dt>
+    <dd>Enables Semgrep Cloud Platform to list repository names on the project setup page.</dd>
+    <dt>Reading the list of organization members</dt>
+    <dd>Enables Semgrep Cloud Platform to determine who can manage your Semgrep organization based on your GitHub organization's members list.</dd>
+    <dt>Reading and writing pull requests</dt>
+    <dd>Enables Semgrep Cloud Platform to comment about findings on pull requests.</dd>
+    <dt>Reading and writing actions</dt>
+    <dd>Enables Semgrep Cloud Platform to cancel stuck jobs, rerun jobs, pull logs from jobs, and perform on-demand scanning.</dd>
+    <dt>Reading <a href="https://docs.github.com/en/rest/reference/checks">GitHub Checks</a></dt>
+    <dd>Facilitates debugging of Semgrep Cloud Platform when configured out of <a href="https://docs.github.com/en/actions">GitHub Actions</a>.</dd>
+    <dt>Reading and writing security events</dt>
+    <dd>Enables integration with GitHub Advanced Security (for example, to show Semgrep results).</dd>
+    <dt>Reading and writing secrets</dt>
+    <dd>Enables automatically adding of the Semgrep Cloud Platform Token to your repository secrets when onboarding projects. Note: Semgrep cannot read the values of your existing or future secrets (only the names).</dd>
+    <dt>Reading and writing 2 files</dt>
+    <dd>Enables Semgrep Cloud Platform to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code> and <code>.semgrepignore</code> files.</dd>
+    <dt>Reading and writing workflows</dt>
+    <dd>Enables Semgrep Cloud Platform to configure itself to run in CI by writing to <code>.github/workflows/semgrep.yml</code>. GitHub allows writing to files within <code>.github/workflows/</code> directory only if this permission is granted along with "Writing a single file."</dd>
+    <dt>Reading and writing pull requests</dt>
+    <dd>Write permissions allow Semgrep Cloud Platform to leave pull request comments about findings. Read permissions allow Semgrep Cloud Platform to automatically remove findings when the pull request that introduced them is closed without merging.</dd>
+</dl>
+
+</TabItem>
+
+<TabItem value='permissions-gitlab'>
+
+#### Permissions for GitLab
+
+Semgrep requires the following permissions (scopes) to enable the authentication of a session:
+
+* `openid`
+* `email`
+* `profile`
+* `API`
+
+</TabItem>
+</Tabs>
+
+### Semgrep versions
+
+Many improvements to the Semgrep Cloud Platform experience only work with up-to-date Semgrep CLI versions. As such, Semgrep Cloud Platform only supports the 10 most recent minor versions of Semgrep CLI. For example, if the latest release was 0.114.0, all versions greater than 0.104.0 are supported, while earlier versions, such as 0.103.0, can be deprecated or can result in failures.
```

Comment:
small nitpick, but can we consider using versions that are a bit less awkward? maybe 1.52.0 - 1.62.0? maybe this doesn't matter, not sure.